### ISSUE TITLE
Connecting to device's local nebula IP

### ISSUE THREAD

**Issue by gilby** [slackhq/nebula#493]
I installed nebula on 3 FreeBSD servers using the port and also on MacBook, and can reach every server from all them, but can't ping or connect to the IP address on the device that IP is on. Is that the expected behavior, just a *BSD thing, or do I have my configuration wrong?

I originally thought using nebula would open up the local network interfaces to the devices on the nebula network, but I can't access those from another server. Was I incorrect on that being possible?

The main issue is that many services only bind to one IP address or all IP addresses, so binding it to the nebula IP means I can't access the server from the server it's running on. I'll get around it by firewalling the access to it so it's only accessible on the nebula and intranet.

**Comment by caguiclajmg** [comment]
> I originally thought using nebula would open up the local network interfaces to the devices on the nebula network, but I can't access those from another server. Was I incorrect on that being possible?

Are you referring to the IP addresses of the hosts that are on a different network interface? No, this isn't a common feature in VPN software

Technically, it is possible, but I would highly suggest against it. If you really want to do this, you can set up static routes on every client that you need to connect from and then use the target's nebula IP as the gateway to reach other interfaces.

> The main issue is that many services only bind to one IP address or all IP addresses, so binding it to the nebula IP means I can't access the server from the server it's running on. I'll get around it by firewalling the access to it so it's only accessible on the nebula and intranet.

Unless you're using software from the stone age, this shouldn't be a problem.

A lot of services allow binding to multiple specific addresses (e.g. in OpenSSH you can specify multiple `ListenAddress` directives).

**Comment by gilby** [comment]
Yes, I was referring to the machine's private address in that datacenter.

The software from the stone age is MariaDB. Looks like MySQL got out of the stone age a couple years ago and allows it, though we're not switching to that.

The main issue remains though and that is of connecting to services on the host's nebula IP.

**Comment by caguiclajmg** [comment]
You still have the option of binding to all interfaces and just firewalling off the addresses you don't want people to see.

If you don't want that, there still are a couple of sane options.

1. NAT a port on your nebula IP to the port on the private interface
2. Set up a tunnel (e.g. frp, ssh)

**Comment by ALDemonPit** [comment]
Nothing

**Comment by brad-defined** [comment]
I spent a bit of time investigating this, thanks for reporting! I found that on Linux devices, packets destined to a service listening on the local Nebula IP are routed through the lo device, and packets flow. On my Mac, packets destined to the local Nebula IP are routed through my Nebula tun device, which drops them. I didn't test on a FreeBSD system, though I suspect it's the same as the Mac.

On Linux, I ran `ip route get <local Nebula IP>` to see which device is used. On my Mac, I ran `route get <local Nebula IP>`.
